hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/managed/ManagedSSTDumpIterator.java

```diff
@@ -0,0 +1,203 @@
+  private Process process;
+  private static final String SST_DUMP_TOOL_CLASS =
+          "org.apache.hadoop.hdds.utils.db.managed.ManagedSSTDumpTool";
+  private static final String PATTERN_REGEX =
+          "'([^=>]+)' seq:([0-9]+), type:([0-9]+) =>";
+
+  public static final int PATTERN_KEY_GROUP_NUMBER = 1;
+  public static final int PATTERN_SEQ_GROUP_NUMBER = 2;
+  public static final int PATTERN_TYPE_GROUP_NUMBER = 3;
+  private static final Pattern PATTERN_MATCHER =
+          Pattern.compile(PATTERN_REGEX);
+  private BufferedReader processOutput;
+  private StringBuilder stdoutString;
+
+  private Matcher currentMatcher;
+  private int prevMatchEndIndex;
+  private KeyValue currentKey;
+  private char[] charBuffer;
+  private KeyValue nextKey;
+
+  private long pollIntervalMillis;
+  private Lock lock;
+
+
+  public ManagedSSTDumpIterator(String sstDumptoolJarPath,
+                                String sstFilePath,
+                                long pollIntervalMillis) throws IOException {
+    File sstFile = new File(sstFilePath);
+    if (!sstFile.exists() || !sstFile.isFile()) {
+      throw new IOException(String.format("Invalid SST File Path : %s",
+              sstFile.getAbsolutePath()));
+    }
+    this.pollIntervalMillis = pollIntervalMillis;
+    this.lock = new ReentrantLock();
+    init(sstFile, sstDumptoolJarPath);
+  }
+
+  private void init(File sstFile, String sstDumptoolJarPath)
+          throws IOException {
+    List<String> args = Lists.newArrayList(
+            "--file=" + sstFile.getAbsolutePath(),
+            "--command=scan");
+    process = HddsServerUtil.getJavaProcess(Collections.emptyList(),
+            sstDumptoolJarPath, SST_DUMP_TOOL_CLASS, args).start();
+    processOutput = new BufferedReader(new InputStreamReader(
+            process.getInputStream(), StandardCharsets.UTF_8));
+    stdoutString = new StringBuilder();
+    currentMatcher = PATTERN_MATCHER.matcher(stdoutString);
+    charBuffer = new char[8192];
+    next();
+  }
+
+  private void checkSanityOfProcess() {
+    if (!process.isAlive() && process.exitValue() != 0) {
+      throw new RuntimeException("Process Terminated with non zero " +
+              String.format("exit value %d", process.exitValue()));
+    }
+  }
+
+  @Override
+  public boolean hasNext() {
+    checkSanityOfProcess();
+    return nextKey != null;
+  }
+
+  @Override
+  public KeyValue next() throws RuntimeIOException {
+    checkSanityOfProcess();
+    try {
+      lock.lock();
+      currentKey = nextKey;
+      nextKey = null;
+      while (!currentMatcher.find()) {
+        try {
+          if (prevMatchEndIndex != 0) {
+            stdoutString = new StringBuilder(stdoutString.substring(
+                    prevMatchEndIndex, stdoutString.length()));
+            prevMatchEndIndex = 0;
+            currentMatcher = PATTERN_MATCHER.matcher(stdoutString);
+          }
+          Thread.sleep(pollIntervalMillis);
```

Comment:
Is this strictly necessary? Is there a better way to achieve the same purpose?

Generally we do want to avoid `sleep` in the prod code.